src/LibVLCSharp.Forms/Shared/PlaybackControls.xaml.cs

```diff
@@ -821,14 +859,126 @@ private void OnAudioTracksChanged(object sender, EventArgs e)
             UpdateAudioTracksSelectionAvailability();
         }
 
+        private void OnVideoTracksChanged(object sender, EventArgs e)
+        {
+            UpdateVideoTracksSelectionAvailability();
+        }
+
         private void OnSubtitlesTracksChanged(object sender, EventArgs e)
         {
             UpdateClosedCaptionsTracksSelectionAvailability();
         }
 
-        private async void AudioTracksSelectionButton_ClickedAsync(object sender, EventArgs e)
+
+        private void TracksButton_Clicked(object sender, EventArgs e)
+        {
+            if (ControlsPanel != null)
+                ControlsPanel.IsVisible = false;
+            var audioTrackSouces = LoadTracks(Manager.Get<AudioTracksManager>());
+            var videoTrackSouces = LoadTracks(Manager.Get<VideoTracksManager>());
+            var subtitleTracksSources = LoadTracks(Manager.Get<SubtitlesTracksManager>());
+            if (AudioTracksListView != null && VideoTracksListView != null && SubtitlesTracksListView != null)
+            {
+                AudioTracksListView.ItemsSource = audioTrackSouces;
+                VideoTracksListView.ItemsSource = videoTrackSouces;
+                SubtitlesTracksListView.ItemsSource = subtitleTracksSources;
+            }
+
+            if (TracksOverLayView != null)
+                TracksOverLayView.IsVisible = true;
+        }
+
+        private void AudioTracksItemTapped(object sender, ItemTappedEventArgs e)
+        {
+            var track = (TrackDescription)e.Item;
+            var manager = Manager.Get<AudioTracksManager>();
+            SelectTrack(manager, track, AudioTracksListView);
+        }
+
+        private void VideoTracksItemTapped(object sender, ItemTappedEventArgs e)
         {
-            await SelectTrackAsync(Manager.Get<AudioTracksManager>(), ResourceManager.GetString(nameof(Strings.AudioTracks)));
+            var track = (TrackDescription)e.Item;
+            var manager = Manager.Get<VideoTracksManager>();
+            SelectTrack(manager, track, VideoTracksListView);
+        }
+
+        private void SubtitlesTracksItemTapped(object sender, ItemTappedEventArgs e)
+        {
+            var track = (TrackDescription)e.Item;
+            var manager = Manager.Get<SubtitlesTracksManager>();
+            SelectTrack(manager, track, SubtitlesTracksListView);
+        }
+
+        private void SelectTrack(TracksManager manager, TrackDescription track, ListView? tracksListview)
+        {
+            var tracks = manager.Tracks;
+            var currentTrackId = manager.CurrentTrackId;
+            if (tracks == null || track.Name.Contains("*"))
+                return;
+            var found = false;
+            var index = 0;
+            foreach (var trackDescription in tracks)
```

Comment:
Seems like this foreach could have been replaced by ` var foundTrack = tracks.FirstOrDefault(t => ...)` and a if/else depending on if it was found.